File: src/types/incident.ts
import { Division, Journal } from "./journal";

export interface Location {
  name: string;
  coordinates: string;
}

export type Incident = {
  id: string;
  name: string;
  createdAt: Date;
  updatedAt: Date;
  deletedAt: Date;
  closedAt: Date;
  location: Location;
  divisions: Division[];
  journals: Journal[];
};

export interface IncidentListData {
  incidents: Incident[];
}

export interface IncidentDetailsData {
  incidents_by_pk: Incident;
}

export interface IncidentDetailsVars {
  incidentId: string;
}


export interface InsertIncidentData {
    insert_incidents_one: Incident
}

export interface InsertIncidentVars {
  name: string
  location: string
  divisions: DivisionInput[]
  journalName: string
}

export interface DivisionInput {
  name: string,
  description: string,
}

File: graph/model/incident.go
package model

import (
	"time"

	"github.com/google/uuid"
)

type Incident struct {
	ID        uuid.UUID  `json:"id"`
	ClosedAt  time.Time  `json:"closedAt"`
	CreatedAt time.Time  `json:"createdAt"`
	UpdatedAt time.Time  `json:"updatedAt"`
	DeletedAt time.Time  `json:"deletedAt"`
	Divisions []Division `json:"divisions"`
	Name      string     `json:"name"`
	Journals  []Journal  `json:"journals"`
	Location  Location   `json:"location"`
}

type NewIncident struct {
	Name     string  `json:"name"`
	Location string  `json:"location"`
	Lat      float64 `json:"lat"`
	Long     float64 `json:"long"`
}
